pipeline/inspection.py

```diff
@@ -54,6 +55,29 @@ def convert_value(cls, v):
             return extract_first_number(v)
         return None
 
+# class syntax
+
+class RegistrationNumberType(str, Enum):
+    """
+    Represents the type of registration number for fertilizers.
+
+    - INGREDIENT: Refers to a registration number associated with a specific ingredient in the ingredient list of a fertilizer.
+    - FERTILIZER: Refers to the unique registration number assigned to the fertilizer product itself.
+    """
+    INGREDIENT = "ingredient_component"
+    FERTILIZER = "fertilizer_product"
+
+class RegistrationNumber(BaseModel):
+    identifier: str = Field(..., description="A string composed of 7-digit number followed by an uppercase letter.")
+    type: Optional[RegistrationNumberType] = None
+
+    @field_validator("identifier", mode="before")
+    def check_registration_number_format(cls, v):
+        if v is not None:
+            pattern = r"^\d{7}[A-Z]$"
+            if re.match(pattern, v):
+                return v
+        raise ValueError(f"Invalid registration number format : {v}")
```

Comment:
Where are we catching this error? Would it cause aborting the program? What was wrong about returning None?